components/common/Header.tsx

```diff
@@ -0,0 +1,142 @@
+import styled from '@emotion/styled';
+import router from 'next/router';
+import React, { ReactElement, useState } from 'react';
+
+import ProfileHover from './ProfileHover';
+
+interface HeaderProps {
+  isLoggedIn: boolean;
+}
+
+function Header({ isLoggedIn = false }: HeaderProps): ReactElement {
+  const [isProfileClicked, setIsProfileClicked] = useState(false);
+
+  function handleLogoClick(e: React.MouseEvent<HTMLButtonElement, MouseEvent>) {
+    e.preventDefault();
+    router.push('/');
+  }
+
+  function handleProfileClick(e: React.MouseEvent<HTMLButtonElement, MouseEvent>) {
+    e.preventDefault();
+    setIsProfileClicked(!isProfileClicked);
```

Comment:
이코드 좋다 ㅎㅎ !로 쓰면 되구나.. 항상 분기처리했는데 ㅋㅋ

Reply:
prevState 기반으로 새 state를 정의할 때는 callback으로 이전 상태를 보장해주는 것이 좋아!

```suggestion
  function handleProfileClick(e: React.MouseEvent<HTMLButtonElement, MouseEvent>) {
    e.preventDefault();
    setIsProfileClicked(isProfileClicked => !isProfileClicked);
```

Reply:
참고로 Suggestion은 `Commit suggestion` 누르면 바로 해당 사항 변경해서 커밋할 수 있음~~ 꿀팁입니다👍

Reply:
와웅... 짱이다

Reply:
우왕 그렇구만 감사합니당~~